openbb_terminal/stocks/discovery/news_sentiment_model.py

```diff
@@ -0,0 +1,65 @@
+"""Onclusive Data Model"""
+
+import logging
+import requests
+import pandas as pd
+
+from openbb_terminal.decorators import check_api_key, log_start_end
+from openbb_terminal.rich_config import console
+from openbb_terminal.core.session.current_user import get_current_user
+
+logger = logging.getLogger(__name__)
+
+@log_start_end(log=logger)
+@check_api_key(["API_ALTHUB_TOKEN"])
+def get_data(
+    ticker: str = "",
+    start_date: str = "",
+    end_date: str = "",
+    date: str = "",
+    limit: int = 100,
+    offset: int = 0,
+) -> pd.DataFrame:
+    headers = {
+        "accept": "application/json",
+        "Authorization": f"token {get_current_user().credentials.API_ALTHUB_TOKEN}",
+    }
+
+    df = pd.DataFrame(data=None)
+
+    query_params = {
+        "all_feilds": 'Flase',
+        "ordering":"-published_on,-share_of_article,-pagerank"
+    }
+    if ticker:
+        query_params["ticker"] = ticker
+    if start_date:
+        query_params["published_on__gte"] = start_date
+    if end_date:
+        query_params["published_on__lte"] = end_date
+
+    if start_date and end_date and not date:
+        if start_date > end_date:
+            console.print("start_date must be lessthan end_date")
+            return df
+
+    if date:
+        query_params["published_on"] = date
+        if start_date:
+            del query_params["published_on__gte"]
+        if end_date:
+            del query_params["published_on__lte"]
+
+    if limit:
+        query_params["limit"] = limit
+    if offset:
+        query_params["offset"] = offset
+
+    response = requests.get(
+        "https://althub-backend.invisagealpha.com/api/OnclusiveSentiment/",
+        headers=headers,
+        params=query_params,
+    ).json()
+    df = pd.DataFrame(data=response["results"])
```

Comment:
This assumes every response will be valid.  When no key is provided, there should be error handling.  For example, without the key defined:

```
API_ALTHUB_TOKEN not defined. Set API Keys in ~/.openbb_terminal/.env or under keys menu.
Error: 'NoneType' object has no attribute 'dropna'
```